completions/gcc

```diff
@@ -1,21 +1,67 @@
 # gcc(1) completion                                        -*- shell-script -*-
 
-_gcc()
-{
-    local cur prev words cword
-    _init_completion || return
+# GCC bash completion for older GCC compilers
 
-    if [[ "$cur" == -* ]]; then
+_gcc_help()
+{
+    if [[ "$2" == -* ]]; then
         cc=$( $1 -print-prog-name=cc1 2>/dev/null )
         [[ $cc ]] || return
         COMPREPLY=( $( compgen -W "$( $cc --help 2>/dev/null | tr '\t' ' ' |\
-            command sed -e '/^  *-/!d' -e 's/ *-\([^][ <>]*\).*/-\1/' )" -- "$cur" ) )
+            command sed -e '/^  *-/!d' -e 's/ *-\([^][ <>]*\).*/-\1/' )" -- "$2" ) )
         [[ $COMPREPLY == *= ]] && compopt -o nospace
     else
         _filedir
     fi
 }
 
+_gcc()
+{
+    local cur prev prev2 words cword argument prefix
+    _init_completion || return
+    _expand || return
+
+    # Test that GCC is recent enough and if not fallback to
+    # parsing of --completion option.
+    gcc --completion=" " 2>/dev/null
+    if [[ "$?" != 0 ]]; then
+      _gcc_help $1 $cur
+      return
+    fi
+
+    # extract also for situations like: -fsanitize=add
+    if [[ $cword > 2 ]]; then
+      prev2="${COMP_WORDS[$cword - 2]}"
+    fi
+
+    # sample: -fsan
+    if [[ "$cur" == -* ]]; then
+      argument=$cur
+    # sample: -fsanitize=
+    elif [[ "$cur" == "=" && $prev == -* ]]; then
+      argument=$prev$cur
+      prefix=$prev$cur
+    # sample: -fsanitize=add
+    elif [[ "$prev" == "=" && $prev2 == -* ]]; then
+      argument=$prev2$prev$cur
+      prefix=$prev2$prev
+    # sample: --param lto-
+    elif [[ "$prev" == "--param" ]]; then
+      argument="$prev $cur"
+      prefix="$prev "
+    fi
+
+    if [[ "$argument" == "" ]]; then
+      _filedir
+    else
+      # In situation like '-fsanitize=add' $cur is equal to last token.
+      # Thus we need to strip the beginning of suggested option.
+      flags=$( gcc --completion="$argument" 2>/dev/null | sed "s/^$prefix//")
```

Comment:
* `flags` should probably be made `local`
* Should invoke `"$1"` instead of `gcc`
* The sed will fail if there are some problematic regex chars in `$prefix`. Would it be possible to just remove (length of characters in `$prefix`) from the start of these strings?
* If continuing to use sed nevertheless, should invoke `command sed` instead of `sed`